### PR TITLE
Added basic show/hide with display value tracking

### PR DESCRIPTION
**Pull request by gricard**
I've added basic $('elem').show() / .hide() support. It keeps track of the previous display value when changing it. It just puts a prop on the elem object to store the old value, similar to what jQuery does, but they put a prop on it to track its key in the private data abstraction they use, which is unnecessary here.

I added tests and updated the documentation also.

### CODE REVIEW

**Comment by gricard** [comment]
@shshaw / @kenwheeler either of you able to take a look at this? :)

**Comment by shshaw** [comment]
Thank you for the pull request! I appreciate the thoroughness in updating the documentation and sizes as well.

Haven't had time to test it, but the code seems a bit verbose overall, and may be increasing the library size quite a bit for functionality that isn't essential. Do you think you can trim it down a bit to prevent bloat?

**Comment by gricard** [comment]
@shshaw Ah, yeah. I noticed that when I updated them. The docs appeared to not have been updated in some time. Right now they state 20K / 9.8K / 3.5K, whereas when I actually download it and build it I see these file sizes:
```
-rw-r--r--   1 gabe  staff  22589 Feb  7 17:11 cash.js
-rw-r--r--   1 gabe  staff  10421 Feb  7 17:14 cash.min.js
-rw-r--r--   1 gabe  staff   3712 Feb  7 17:11 cash.min.js.gz
```

...which translates to 22K / 10.2K / 3.6K. 

My changes result in dist files of these sizes:
```
-rw-r--r--   1 gabe  staff  23688 Jan 25 06:51 cash.js
-rw-r--r--   1 gabe  staff  10860 Feb  7 17:15 cash.min.js
-rw-r--r--   1 gabe  staff   3876 Jan 25 06:51 cash.min.js.gz
```

...which translate to 23.1K / 10.6K / 3.8K.

That's a fairly minor bump in size, IMO. The only way I see to trim it down is to get rid of the tracking for the prior value of the display attribute, which would result in buggy show/hide behavior. It's up to you guys.

**Comment by fabiospampinato** [comment]
I don't think we should add these functions, generally you should do something like this instead:

```css
.hidden {
  display: none !important;
}
```

```js
$.fn.hide = function () {
  return this.addClass ( 'hidden' );
};
$.fn.show = function () {
  return this.removeClass ( 'hidden' );
};
```

**Comment by gricard** [comment]
I think the reason that jQuery, etc. do it the way I did in my patch is because there are no side-effects when directly manipulating the display attribute that way. If you add a CSS class, then you’re potentially causing conflicts with applications that use this library.

**Comment by fabiospampinato** [comment]
@gricard I'll consider this PR, but I have a few requests first:

1. Can you rebase and write your code in a way that's more aligned with the rest of the code?
2. I haven't actually checked how many bytes this will add to our minified and gzipped bundle, can you make sure those are as few as possible?
3. Can you also implement a very basic version of `$.fn.toggle`? I think we should add this one too for completeness.
4. Can you add some tests?

Thank you.

**Comment by gricard** [comment]
1. I don’t understand what you’re asking for that. How is it out of alignment with the rest of the code? 

2. I did check. There’s a comment about it above. It’s negligible. I’m not sure if it can be reduced further. 

3. What is $.fn.toggle?

4. I did. Are there any other specific tests you’re looking for?

TBH, it’s been so long since I posted this PR that I’ve worked around my need for it anyway. I’m not likely going to invest any more time in it. If it doesn’t get merged, it’s no skin off my back.

**Comment by fabiospampinato** [comment]
> I don’t understand what you’re asking for that. How is it out of alignment with the rest of the code?

For starters there's no longer a `src/css.js` file. Each collection method has its own file and helper functions are in their `helper` folders. Also your coding style is different from the one used in rest of the code.

> I did check. There’s a comment about it above. It’s negligible. I’m not sure if it can be reduced further.

I think you did the check on the old repo, the results won't necessarily be the same with the current state of the repo.

I think there's probably some room for optimizations anyway. You're writing `elem[displayTrackerProp] ? elem[displayTrackerProp] : "";` instead of `elem[displayTrackerProp] || '';` and the `updates` array seems unnecessary to me.

> What is $.fn.toggle?

Basically [this](http://api.jquery.com/toggle/), but without animations. If `hide` is `removeClass`, and `show` is `addClass`, then `toggle` is `toggleClass`.

> I did. Are there any other specific tests you’re looking for?

You're right, I missed those, sorry about that.

> TBH, it’s been so long since I posted this PR that I’ve worked around my need for it anyway. I’m not likely going to invest any more time in it. If it doesn’t get merged, it’s no skin off my back.

I'm closing this then. If you change your mind feel free to resubmit another PR.
